src/core/build_target.go

```diff
@@ -140,6 +141,8 @@ type BuildTarget struct {
 	Progress float32 `print:"false"`
 	// The results of this test target, if it is one.
 	Results TestSuite `print:"false"`
+	// A mutex to control access to Results
+	ResultsMux sync.Mutex
```

Comment:
You never seem to actually call this? 

Would also suggest maybe making it private and having some method to control access to the results.

Reply:
It's called in test_step.go before doing anything to the Results field. I could move all the methods for updating the results to this file and make the mux private.